packages/waku/src/router/base-types.ts

```diff
@@ -1,6 +1,13 @@
+import type {
+  PagePath,
+  PropsForPages,
+} from './create-pages-utils/inferred-path-types.js';
+
 export type { PathsForPages } from './create-pages-utils/inferred-path-types.js';
 
 // eslint-disable-next-line @typescript-eslint/no-empty-object-type
 export interface RouteConfig {
   // routes to be overridden by users
 }
+
+export type PageProps<Path extends PagePath<RouteConfig>> = PropsForPages<Path>;
```

Comment:
This util is specifically for `createPages`, right? Can you add a comment to clarify it?

Reply:
added the comment. 

I looked at `defineEntries` and that could also use this same type system in theory. Happy to work on that soon if you think that'd be helpful.

Reply:
I'm not sure what it's about `defineEntries`. If it's just simply improve type inference, it would be nice. If it's about manual type annotation, it's not very important for now.

Reply:
I think both might be easy enough to look into. I think most of the hard work was done already for `createPages` and might be easily shareable with other things like `defineEntries`.

Reply:
> might be easily shareable with other things like `defineEntries`.

idea can be shareable, but not the code. I don't think manual annotation for `defineEntries` is necessary, just type inference would be good.